lib/ansible/playbook/task.py

```diff
@@ -215,7 +215,7 @@ def preprocess_data(self, ds):
             if k in ('action', 'local_action', 'args', 'delegate_to') or k == action or k == 'shell':
                 # we don't want to re-assign these values, which were determined by the ModuleArgsParser() above
                 continue
-            elif k.replace("with_", "") in lookup_loader:
+            elif k.startswith('with_') and k.replace("with_", "") in lookup_loader:
```

Comment:
Why is this change needed for the addition of the var lookup? I don't understand the context, any chance we could add it to the comment?

Reply:
vars let me discover we were doing something improper and allowing direct use of the lookup as a keyword, since 'vars' already exists ... it was getting overloaded, this fixes that bug.

Reply:
:+1: